bundles/org.openhab.core.io.transport.mdns/src/main/java/org/openhab/core/io/transport/mdns/internal/MDNSClientImpl.java

```diff
@@ -277,7 +281,55 @@ private void createJmDNSByAddress(InetAddress address) {
     @Override
     public void onChanged(List<CidrAddress> added, List<CidrAddress> removed) {
         logger.debug("ip address change: added {}, removed {}", added, removed);
-        close();
-        start();
+
+        // First check if there is really a jmdns instance to remove or add
+        boolean changeRequired = false;
+        for (CidrAddress address : removed) {
+            JmDNS jmdns = jmdnsInstances.get(address.getAddress());
+            if (jmdns != null) {
+                changeRequired = true;
+                break;
+            }
+        }
+        if (!changeRequired) {
+            for (CidrAddress address : added) {
+                JmDNS jmdns = jmdnsInstances.get(address.getAddress());
+                if (jmdns == null) {
+                    changeRequired = true;
+                    break;
+                }
+            }
+        }
+        if (!changeRequired) {
+            logger.debug("mDNS services already OK for these ip addresses");
+            return;
+        }
+
+        for (ServiceDescription description : activeServices) {
+            unregisterServiceInternal(description);
+        }
+        for (CidrAddress address : removed) {
+            JmDNS jmdns = jmdnsInstances.remove(address.getAddress());
+            if (jmdns != null) {
+                closeQuietly(jmdns);
+                logger.debug("mDNS service has been stopped ({})", jmdns.getName());
+            }
+        }
+        for (CidrAddress address : added) {
+            JmDNS jmdns = jmdnsInstances.get(address.getAddress());
+            if (jmdns == null) {
+                createJmDNSByAddress(address.getAddress());
```

Comment:
I think the issue with this change is that `added` contains any network interface and not the filtered interfaces provided by `MDNSClientImpl#getAllInetAddresses()` causing #3976.

Reply:
Yes, you are right, we should ignore here added network interface addresses not matching the OH network configuration.
I am sorry to not think about that when providing this fix.